scriptworker/cot/verify.py

```diff
@@ -1030,11 +1041,77 @@ def verify_link_in_task_graph(chain, decision_link, task_link):
     }
 
 
-def _get_additional_github_releases_jsone_context(parent_link, decision_link):
+async def _get_additional_github_releases_jsone_context(decision_link):
+    context = decision_link.context
+    source_env_prefix = context.config['source_env_prefix']
+    task = decision_link.task
+    repo_url = get_repo(task, source_env_prefix)
+    repo_owner, repo_name = extract_github_repo_owner_and_name(repo_url)
+    tag_name = get_revision(task, source_env_prefix)
+
+    all_releases = await load_json_or_yaml_from_url(
+        context,
+        url=get_all_releases_url(context, repo_owner, repo_name),
+        path=os.path.join(
+            context.config['work_dir'], '{}_all_github_releases.json'.format(decision_link.name)
+        )
+    )
+    release_matching_tag = get_single_item_from_sequence(
+        all_releases,
+        condition=lambda release: release.get('tag_name') == tag_name,
+    )
+    release_id = release_matching_tag['id']
+    release_data = await load_json_or_yaml_from_url(
+        context,
+        url=get_release_url(context, repo_owner, repo_name, release_id),
+        path=os.path.join(
+            context.config['work_dir'], '{}_releases_{}.json'.format(decision_link.name, release_id)
+        )
+    )
+
+    # The release data expose by the API[1] is not the same as the original event[2]. That's why
+    # we have to rebuild the object manually
+    #
+    # [1] https://developer.github.com/v3/repos/releases/#get-a-single-release
+    # [2] https://developer.github.com/v3/activity/events/types/#releaseevent
+    return {
+        'event': {
+            'repository': {
+                # TODO: Append ".git" to clone_url in order to match what GitHub actually provide.
+                # This can't be done at the moment because some mobile projects still rely on the
+                # bad value
+                'clone_url': repo_url,
+                'html_url': repo_url,
+            },
+            'release': {
+                'tag_name': tag_name,
+                'target_commitish': release_data['target_commitish'],
+                # Releases are supposed to be unique. Therefore, we're able to use the latest
+                # timestamp exposed by the API (unlike PRs, for instance)
+                'published_at': release_data['published_at'],
+            },
+            'sender': {
+                'login': release_data['author']['login'],
+            },
+        },
+    }
+
+
+def _get_additional_git_cron_jsone_context(decision_link):
     source_env_prefix = decision_link.context.config['source_env_prefix']
     task = decision_link.task
     repo = get_repo(task, source_env_prefix)
+
+    # TODO remove the call to get_triggered_by() once Github repos don't define it anymore.
+    user = get_triggered_by(task, source_env_prefix)
+    if user is None:
+        # We can't default to 'cron' (like in hg) because https://github.com/cron is already taken
+        user = 'TaskclusterHook'
```

Comment:
Is there any reason we can't have `None` here?

Can you explain a little bit how this is used, and what the context is? If we aren't using `cron` because that already exists, I worry if someone were to create this user, they might be able to subvert some part of our checks. We could, of course, create the user, but I wonder if there is some way to structure things to avoid that.

Reply:
Yeah, we need to provide a user name so `metadata.owner` is defined in the task definition. `metadata.owner` must be an email address. For instance: https://github.com/mozilla-mobile/fenix/blob/b8ab51318e99706656f54669594ef8263cf560b6/.taskcluster.yml#L114

This value is hardcoded in the scripts that spin the cron graphs. For instance: https://github.com/mozilla-mobile/fenix/blob/b8ab51318e99706656f54669594ef8263cf560b6/automation/taskcluster/schedule_nightly_graph.py#L64

An alternative solution could be to use `cron@noreply.mozilla.org` instead of an `@users.noreply.github.com` address.

Reply:
If this is purely a metadata thing, using a Mozilla email address seems good. Non-blocker.

Reply:
I thought about this -- since we have a limited number of repos that run cron, we can change the email in `.taskcluster.yml` as soon as the new email is rolled out in `cot.verify`, and then we should be good.